dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerAccessManager.java

```diff
@@ -77,16 +77,35 @@ public class DefaultTrackerAccessManager implements TrackerAccessManager {
    */
   @Override
   public List<String> canRead(@Nonnull UserDetails user, TrackedEntity trackedEntity) {
-    // always allow if user == null (internal process) or user is superuser
     if (user.isSuper() || trackedEntity == null) {
       return List.of();
     }
 
-    return canRead(user, trackedEntity, programService.getAllPrograms());
+    return canRead(user, trackedEntity, programService.getAllPrograms(), false);
+  }
+
+  /**
+   * Check the data read permissions and ownership of a tracked entity given the programs for which
+   * the user has metadata access to. Ownership validations will be skipped if the flag is set to
+   * true.
+   *
+   * @return No errors if a user has access to at least one program
+   */
+  @Override
+  public List<String> canRead(
+      @Nonnull UserDetails user, TrackedEntity trackedEntity, boolean skipOwnershipCheck) {
```

Comment:
These `skipOwnershipCheck` are a bit scary to me 😅 I would prefer if `canRead` contains the logic of "user x is an admin so I don't need to check ownership, ..."

Reply:
I don't like it either, but the logic to decide if we can skip the check or not, is to validate the org unit mode is `ALL`. That's fine when using the exporter.
However, the importer doesn't use org unit modes, that's why I went for this solution.
Again, this is temporary and as soon as I'm done with the tickets that need to be merged before the release, I will remove this validation and add it into the query.